### ISSUE TITLE
EnC analyzer reports duplicate semantic diagnostics

### ISSUE THREAD

**Issue by tmat** [dotnet/roslyn#36061]
1) Debug the following console app 
```C#
		static void Main(string[] args)
		{
		    int a = 10; // breakpoint here
		}
```

2) Change the code to:
```C#
		static void Main(string[] args)
		{
		    int a = 132408290348290;
		}
```
		
3) Press F10.

The error list contains duplicate semantic errors:

![image](https://user-images.githubusercontent.com/41759/58601318-19dc6180-823d-11e9-87b4-a03a1291d2d9.png)

This is because the EnC analyzer reports all diagnostics returned by EmitDifference and some of them (syntactic and binding) have also been reported by the diagnostic analyzer. The solution would be to only report emit diagnostics from EnC. But the compiler currently does not tag diagnostics with the phase they originated from.

See also https://github.com/dotnet/roslyn/issues/1400#issuecomment-83884093 and https://github.com/dotnet/roslyn/issues/3675#issuecomment-163436367

An alternative solution would be to call `compilation.GetDiagnostics()` and if there are any, do not attempt to emit and do not report any diagnostics (they were already reported by the diagnostic analyzer). However, this would significantly affect performance of applying the change. This operation is blocking - the customer can't continue debugging without changes applied, so doing as little work as possible is critical. The only diagnostics that need to block change application are those that prevent emitting the symbols affected by the change. Diagnostics on any other symbols can be ignored.

**Comment by tmat** [comment]
Doesn't repro anymore. The diagnostics have been dedup'd.
